### ISSUE TITLE
onChange not working   (＊0＊;) 

### ISSUE THREAD

**Issue by daimajia** [one-signal/emoji-picker#4]
It can not set the originalText value.

**Comment by daimajia** [comment]
https://github.com/one-signal/emoji-picker/blob/gh-pages/lib/js/jquery.emojiarea.js#L412-L414

**Comment by daimajia** [comment]
It's pretty weird.

the following may work for the input text (not for textarea). 
```javascript
this.$textarea.attr('value', this.val()).trigger('change');
```

I have no idea why the `val()` function not work...

**Comment by jasonpang** [comment]
Hey daimajia,

I'm not able to reproduce the issue. Here's what I did to try to reproduce the issue:

1. I opened my locally hosted version of the demo page.
2. I added console logging to `jquery.emojiarea.js:onChange()` (lines 412-415):

    ```
    EmojiArea_WYSIWYG.prototype.onChange = function(e) {
    	console.log('(Before) this.$textarea.val(): ', this.$textarea.val());
    	this.$textarea.val(this.val()).trigger('change');
    	console.log('(After) this.$textarea.val(): ', this.$textarea.val());
    };
    ```
3. I open my local demo page and type "1", "2", "3", "4", "5" in a sequence.

This is the result I get:

![image](https://cloud.githubusercontent.com/assets/1015970/9614658/d908fb20-50a7-11e5-8a2e-403e792f9d6a.png)

It looks like that line is working?

**Comment by daimajia** [comment]
There is an original-input element (`this.$textarea`). When I input something, the original-input still have no value. I don't if I maybe misunderstand the mechanism. How can I guarantee the original-input value is always equal to `this.val()`? 


![image](https://cloud.githubusercontent.com/assets/2503423/9620813/f350382c-5151-11e5-9e5f-905dd6c92d3b.png)

**Comment by jasonpang** [comment]
A couple things:

1. If you're inspecting the input field in Chrome's Developer Tools, I think you'll find that input fields will not have their contents displayed. I'm not sure why this is the case; it's just not displayed. You'll instead have to call `$('<input field>').val()` to get the value.

2. Right now, the original input field's value gets updated on the emoji rich area's [`change`](https://developer.mozilla.org/en-US/docs/Web/Events/change) event. `change` fires when the value entered into the emoji rich area's field is "committed". An example of committing is clicking away from the emoji rich area after entering text.

However, if you change `emoji-picker.coffee:21` to read `norealTime: false`, the original input area's value will be updated on each `keyPress` event, which I think is what you want to get real time updates.

In summary, inspecting the input field in Chrome's Dev Tools doesn't show the value of input fields (I think), and if you want the original input field to be updated after each keystroke, you can change `emoji-picker.coffee:21` to read `norealTime: false`.

**Comment by gdeglin** [comment]
Closing this for now. Let us know if you're still experiencing any issues @daimajia.

**Comment by daimajia** [comment]
@jasonpang  @gdeglin  Thanks so much.
